src/server/gc_worker/compaction_filter.rs

```diff
@@ -279,10 +297,8 @@ impl CompactionFilter for WriteCompactionFilter {
                 WriteType::Put => self.remove_older = true,
                 WriteType::Delete => {
                     self.remove_older = true;
-                    if self.bottommost_level {
-                        filtered = true;
-                        self.gc_before_delete_mark(key, key_prefix);
-                    }
+                    filtered = true;
+                    self.gc_before_delete_mark(key, key_prefix);
```

Comment:
not related to this PR, but here we trigger gc_before_delete_mark once we hit a deletion marker. Can we trigger it only when we pass the oldest version of the current key in the current compaction?